Sources/Sentry/SentryHttpTransport.m

```diff
@@ -84,36 +92,47 @@ - (void)sendEnvelope:(SentryEnvelope *)envelope
 // flow
 - (void)sendAllCachedEnvelopes
 {
-    if (![self.requestManager isReady]) {
-        return;
+    @synchronized(self) {
```

Comment:
This is only called in a background thread, right?

Reply:
Nope, the first time it is called from `sendEnvelope` and this could also be the main/UI thread.

Reply:
Task scheduled to follow up on this